metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraAspectDao.java

```diff
@@ -559,8 +562,13 @@ private SimpleStatement generateSaveStatement(EntityAspect aspect, boolean inser
 
       Update u = uwa.whereColumn(CassandraAspect.URN_COLUMN).isEqualTo(literal(aspect.getUrn()))
               .whereColumn(CassandraAspect.ASPECT_COLUMN).isEqualTo(literal(aspect.getAspect()))
-              .whereColumn(CassandraAspect.VERSION_COLUMN).isEqualTo(literal(aspect.getVersion()))
-              .ifExists();
+              .whereColumn(CassandraAspect.VERSION_COLUMN).isEqualTo(literal(aspect.getVersion()));
+
+      if (updateIfCreatedOn != null) {
```

Comment:
Is it feasible for `updateIfCreatedOn == 0` here? I'm not sure where `insert` gets set so I can't reason through this properly. Maybe @RyanHolstien will have an opinion?

Reply:
If it's 0 it would go into the above block, not this one. Insert gets set to true when no previous version exists.